test/cpp/jit/test_interpreter.cpp

```diff
@@ -1,9 +1,114 @@
 #include "test/cpp/jit/test_base.h"
 #include "test/cpp/jit/test_utils.h"
 
+#include <stdexcept>
 namespace torch {
 namespace jit {
 
+void testTypeCheck() {
+  {
+    auto graph = std::make_shared<Graph>();
+    std::unordered_map<std::string, Value*> vmap;
+    parseIR(
+        R"IR(
+graph(%a.1 : Tensor,
+      %b.1 : Tensor):
+  %t : Float(2:2, 2:1, device=cuda, requires_grad=1), %t0 : Float(3:3, 3:1), %type_matched : bool = prim::TypeCheck(%a.1, %b.1)
+  %14 : Tensor = prim::If(%type_matched)
+    block0():
+      -> (%a.1)
+    block1():
+      -> (%b.1)
+  return (%14)
+  )IR",
+        &*graph,
+        vmap);
+
+    Code print_function(graph, "");
+
+    auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 1);
+    auto a = at::zeros({2, 2}, options);
+    auto b = at::ones({3, 3});
+    auto a2 = at::ones({2, 2});
+    auto a3 = at::ones({2, 2}); //.idtype(at::kInt);
+    a3 = a3.to(at::kInt);
+
+    a.set_requires_grad(true);
+    a2.set_requires_grad(false);
+
+    {
+      // TypeCheck yields to true! Tensor a is returned
```

Comment:
I'm not sure I understand how the tests work: shouldn't we be checking the return value of the op (true/false)? The tensors should be unchanged before and after the call, or am I missing something? Also, it could be useful to check the entire content of the stack after the op and not just a single element (at least in one test).

Reply:
I updated the test. Hope it clearer